File: Trees/BinaryTrees/PathFromRootToTarget.java
package Trees.BinaryTrees;

import java.util.ArrayList;

public class PathFromRootToTarget {

    private static boolean findPathFromRootToTarget(TreeNode root, int target, ArrayList<Integer> ans) {
        if (root == null) return false;

        ans.add(root.data);
        if (root.data == target) return true;

        if (findPathFromRootToTarget(root.left, target, ans) ||
                findPathFromRootToTarget(root.right, target, ans)) return true;

        ans.remove(ans.size() - 1);
        return false;
    }

    public static void main(String[] args) {
        ArrayList<Integer> arr = new ArrayList<>();
        findPathFromRootToTarget(TreeNode.getInstance1(), 4, arr);
        for (int i : arr) System.out.print(i + " ");
    }
}

File: Trees/BinaryTrees/PathFromRootToTarget.cpp
#include <bits/stdc++.h>
using namespace std;

typedef struct TreeNode{
    int val;
    TreeNode* left,*right;

    TreeNode(){
    }

    TreeNode(int val){
        this->val = val;
        left = right = NULL;
    }
}TreeNode;

class Util{
	public:
    static TreeNode* getTree(){
        TreeNode* root = new TreeNode(10);
        root->left = new TreeNode(20);
        root->right = new TreeNode(30);

        root->left->left = new TreeNode(40);
        root->left->right = new TreeNode(50);
        root->left->right->left = new TreeNode(60);
        root->left->right->right = new TreeNode(70);
        return root;
    }
    static bool printPathFromRootToTarget(TreeNode* root,int target,vector<int>&path){
        if(root == nullptr) return false;
        path.push_back(root->val);
        if(root->val == target) return true;
        if(printPathFromRootToTarget(root->left,target,path) ||  printPathFromRootToTarget(root->right,target,path))  return true;
        path.pop_back();
        return false;
    }
};
int main(){
	vector<int> path;
    Util::printPathFromRootToTarget(Util::getTree(),70,path);
    for(int i : path)	cout<<i<<" ";
    return 0;
}